src/Sensor/SensorBms.cpp

```diff
@@ -0,0 +1,144 @@
+#include "SensorBms.h"
+
+#define TEMP_ID_INTERNAL    0x00
+#define TEMP_ID_BATTERY_1   0x01
+#define TEMP_ID_BATTERY_2   0x02
+
+#define REQ_DATA_LENGTH     8
+
+#define RSP_STATUS_BYTE     0x0
+#define RSP_PARAM_ID_BYTE   0x1
+#define RSP_DATA_BYTE       0x2
+
+SensorBms::SensorBms(CanInterface &canInterface, uint16_t requestIntervalMs) 
+    : CanListener(canInterface, CAN_BMS_RESPONSE), _requestIntervalMs(requestIntervalMs) {}
+
+void SensorBms::handle() {
+    if(millis() - _lastValidTime >= _requestIntervalMs) {
+
+        for(int i = 0; i < NUM_PARAMS; i++) {
+            CanMessage msg = CAN_MESSAGE_NULL;
+            msg.id = CAN_BMS_REQUEST;
+            msg.dataLength = REQ_DATA_LENGTH;
+            msg.data[0] = _paramIds[i];
+            
+            _canInterface.sendMessage(msg);
+        }
+        
+        _lastValidTime = millis();
+    }
+}
+
+String SensorBms::getHumanName() {
+    return "BMS";
+}
+
+void SensorBms::update(CanMessage message) {
+    DEBUG_SERIAL_LN("RECEIVED RESPONSE FROM BMS");
+
+    if(message.data[RSP_STATUS_BYTE] != TRUE) {
+        DEBUG_SERIAL_LN("Poor Data Received");
+    }
+    else {
+        unsigned statusCode = 0;
+        switch (message.data[RSP_PARAM_ID_BYTE]) {
+            case PARAM_ID_BATTERY_VOLTAGE:
+                _batteryVoltage = parseFloat(message.data + RSP_DATA_BYTE);
+                break;
+            case PARAM_ID_BATTERY_CURRENT:
+                _batteryCurrent = parseFloat(message.data + RSP_DATA_BYTE);
+                break;
+            case PARAM_ID_MAX_CELL_VOLTAGE:
+                _cellVoltageMax = parseFloat(message.data + RSP_DATA_BYTE);    
+                break;
+            case PARAM_ID_MIN_CELL_VOLTAGE:
+                _cellVoltageMin = parseFloat(message.data + RSP_DATA_BYTE);    
+                break;
+            case PARAM_ID_STATUS:
+                statusCode = (message.data[3] << 8) + message.data[2];
+                if(statusCode == STATUS_CHARGING) {
+                    _bmsStatus = "BMS Charging...";
+                    DEBUG_SERIAL_LN("BMS Charging...");
```

Comment:
I think these status debug statements can be removed now that you're done the class